torch/csrc/utils/python_arg_parser.cpp

```diff
@@ -420,7 +437,7 @@ auto FunctionParameter::check(PyObject* obj, std::vector<py::handle> &overloaded
       // if a size is specified (e.g. IntArrayRef[2]) we also allow passing a single int
       return size > 0 && THPUtils_checkLong(obj);
     }
-    case ParameterType::FLOAT_LIST: return (PyTuple_Check(obj) || PyList_Check(obj));
+    case ParameterType::FLOAT_LIST: return is_float_list(obj);
```

Comment:
so this will return true for a list consisting of a mixture of float and complex numbers? I guess you are checking them later, but the name is misleading.

Reply:
Renamed `is_float_list`->`is_float_or_complex_list`